### ISSUE TITLE
OOBE when new user is created

### ISSUE THREAD

**Issue by mrcrutch** [Disassembler0/Win10-Initial-Setup-Script#201]
When creating a new user, OOBE asks for some settings (telemetry related).
Would be possible to hide that to end user and force default settings?

**Comment by Disassembler0** [comment]
> Would be possible to hide that to end user and force default settings?

Short answer: No.

Long answer: Yes, but **it's dangerous**, as it requires use of *sysprep*, which basically **modifies your whole system image**.

There's a documentation for [customizing OOBE](https://docs.microsoft.com/en-us/windows-hardware/customize/desktop/customize-oobe) and [automating OOBE](https://docs.microsoft.com/en-us/windows-hardware/customize/desktop/automate-oobe). This particular settings is the [ProtectYourPC](https://docs.microsoft.com/en-us/windows-hardware/customize/desktop/unattend/microsoft-windows-shell-setup-oobe-protectyourpc) part of OOBE automation. So what you need to do is to create `unattended.xml` file with the OOBE setup according to your liking. This you can do either manually or via *Windows System Image Manager (SIM)* which a part of [Windows ADK](https://docs.microsoft.com/en-us/windows-hardware/get-started/adk-install). If you choose to do it manually, [here](https://www.tenforums.com/installation-upgrade/93862-oobe-sysprep-answer-file-win10-version-1703-a.html) and [here](https://www.tenforums.com/installation-upgrade/127414-automate-just-oobe-setup-pass-windows-10-autounattend-xml.html) are some nice examples on which your config can be based. This `unattended.xml` file is then normally saved in `install.wim` and copied to the online image into `C:\Windows\Panther`, along with *action queues* (that's actually the part which is hard to modify manually) in which the individual sysprep steps are registered.

An example of `unattend.xml` setup fulfilling your goal would be
```xml
<?xml version="1.0" encoding="utf-8"?>
<unattend xmlns="urn:schemas-microsoft-com:unattend">
    <settings pass="oobeSystem">
        <component name="Microsoft-Windows-Shell-Setup" processorArchitecture="amd64" publicKeyToken="31bf3856ad364e35" language="neutral" versionScope="nonSxS" xmlns:wcm="http://schemas.microsoft.com/WMIConfig/2002/State" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
            <OOBE>
                <SkipMachineOOBE>true</SkipMachineOOBE>
                <SkipUserOOBE>true</SkipUserOOBE>
                <HideEULAPage>true</HideEULAPage>
                <HideLocalAccountScreen>true</HideLocalAccountScreen>
                <HideOEMRegistrationScreen>true</HideOEMRegistrationScreen>
                <HideOnlineAccountScreens>true</HideOnlineAccountScreens>
                <HideWirelessSetupInOOBE>true</HideWirelessSetupInOOBE>
                <ProtectYourPC>3</ProtectYourPC>
                <UnattendEnableRetailDemo>false</UnattendEnableRetailDemo>
            </OOBE>
        </component>
    </settings>
</unattend>
```

You'd save this as e.g. `C:\unattend.xml` and then run command
```
C:\windows\system32\sysprep\sysprep.exe /oobe /generalize /reboot /unattend:C:\unattend.xml
```

This would create the appropriate action queues leading to your customizations and then restarted the computer. After next startup, general sysprep steps, including OOBE with your customizations, would be reapplied as if you'd just installed fresh system or applied a major upgrade. This will also ~~fuck up~~ *reinitialize* some of your settings, so generally, I recommend against doing that on already installed, configured and running systems.

There may be a way how to plant the OOBE setup and action queues onto online image even without sysprep, but once again, it's dangerous and may create conflicts with existing configuration. I haven't tried it and and given the complexity of the resulting tweak, I don't intend to.

Bottom line: If you want to customize on large scale, you'll end up creating custom images with SIM and using sysprep anyway. If you want to customize only a few users, you're better off just clicking the button and accepting the defaults, which you'll afterwards customize via tweaks from this repository.
